### DISCUSSION TITLE
Linux Cluster

### DISCUSSION THREAD

**Discussion by KhaledElwazan** [gap-system/gap#5174] [Q&A] [answered]
Does gap support running on Linux clusters?

**Comment by ChrisJefferson** [comment, marked as answer]
GAP can certainly be run on clusters of Linux computers -- I have solved problems which required distributing GAP over around 60 machines.

However, there a number of different types of "linux clusters", and depending on your exact cluster GAP may, or may not, have special support for it. Could you give us an idea of the types of problems you want to solve, and exactly what type of cluster you would be using?

**Reply by KhaledElwazan** [reply]
This [link](https://hpc.bibalex.org/about#clusters) provides the details to the machine in question.

**Reply by ChrisJefferson** [reply]
Assuming the cluster can run linux programs, then GAP should work on it fine. It won't take advantage of the GPUs, as GAP contains no code that uses GPUs, and GAP is not multi-threaded (there is HPC-GAP, but I would not recommend it's use unless you are already a GAP expert, as it is not well supported), so you would have to run many copies of GAP to make best use of all the CPU cores, which means you will have to figure out how to split up your problem so it can run in many different copies of GAP.

When I have done parallel work in the past I have often wanted to solve many small problems (like solving a problem for all small groups for example), so my problem naturally split into many pieces which I could then distribute across a large number of copies of GAP running on many different computers.

**Reply by KhaledElwazan** [reply]
I actually did that trick of running multiple instances of GAP on my personal computer to speed a very long loop, using some bash scripts and python.  It worked, surprisingly, well. It will be interesting to kick it up a notch on the HPC.
Thank you so much for your elaborate answer.
